packages/store/src/providers.ts

```diff
@@ -12,6 +14,12 @@ import type { Awareness } from 'y-protocols/awareness';
  */
 export interface SyncProvider {
   awareness?: Awareness;
+  /**
+   * Each {@link SyncProvider} is essentially used as a mirror in practice, so for now,
+   * the easiest approach is to simply enable every sync provider to provide their own
+   * separate blob storage option.
+   */
+  blobStorage?: BlobStorage;
```

Comment:
I think `BlobStorage` can be separated from the SyncProvider, since I consider the SyncProvider should only be be used to sync YDoc (via different databased or network protocols).